### ISSUE TITLE
Compile BAGIS add-ins for ArcGIS 10.7

### ISSUE THREAD

**Issue by lbross** [PSU-CSAR/vb-bagis#75]
@jduh to configure a system for @lbross to RDP to for a 10.7 development environment. The following components will need to be installed by someone with admin access:

1. VS Community 2017: 10.7 works with 2015 and 2017 so we might as well go with the latest. I am already using this version for Pro and Python development
2. ArcMap 10.7 and ArcMap SDK 10.7
3. Git For Windows 2.0 
4. TortoiseGit

**Comment by jdduh** [comment]
While David is setting up the new computer with ArcGIS 10.7.1, the original RDP computer (ggr-ch-424H-G0) is still available for 10.5.1. The computer probably will be renamed and kept as a time-capsule for BAGIS addin development on 10.5.1.

**Comment by lbross** [comment]
I have a 10.5.1 environment on my PSU laptop. Let me know when I should upgrade to a newer version and we can use the original RDP computer for maintaining a 10.5.1 branch.

**Comment by jdduh** [comment]
I have installed the requested software packages on a new computer and will send you the name of the computer by email. The computer still needs to be configured so that it can be waken up by LAN or won't go to sleep.

**Comment by lbross** [comment]
There is a BAGIS add-in for v10.7 posted on basins in the For ArcGIS 10.7.1 folder. There were a few libraries that had the specific version property set to 'true'. This might be why 10.7.1 could not use the 10.5.1 version that we are currently releasing. I also upgraded the .NET framework to 4.5.2 so that it is the same as ArcGIS 10.7.1. Previously it had been 4.5. This caused some issues with calls to the Windows.Forms classes. 4.5.2 requires them to be fully qualified as System.Windows.Forms.

Overall, pretty smooth transition and the performance feels much improved. 10.5.1 was slow when connecting with ArcCatalog for the first time during a session. Please test this add-in. If you find that it works, I will build BAGIS-H and BAGIS-P for 10.7.1.

**Comment by lbross** [comment]
I also updated the default settings file on basins to point at the pourpoint, snotel, and snow course services on NRCS ArcGIS online. I used these settings to validate the add-in for v10.7.

**Comment by jdduh** [comment]
The add-in runs well in 10.7.1. Please work on BAGIS-H and -P when you get a chance. You might want to put a comment in the add-in somewhere so that users can easily verify which ArcGIS version the addin is compatible with. Maybe here?

![image](https://user-images.githubusercontent.com/11510708/65903954-467ec200-e372-11e9-8c3a-401659f4092c.png)

**Comment by lbross** [comment]
Tried to upgrade BAGIS-H today and it seems there is a bug with the ZonalStatisticsAsTable tool. This manifests itself when trying to assign sequential HRU numbers to an AOI after running the elimination tool. Here is the error:
Exception thrown: 'System.Runtime.InteropServices.COMException' in ESRI.ArcGIS.Geoprocessor.dll
GP error: Executing: ZonalStatisticsAsTable C:\AOIs\animas_AOI_prms\zones\elim_2\elim_2.gdb\grid_v grid_code C:\AOIs\animas_AOI_prms\surfaces.gdb\flow_accumulation C:\AOIs\animas_AOI_prms\zones\elim_2\elim_2.gdb\streamOrderTbl DATA MAXIMUM
GP error: Start Time: Mon Sep 30 15:00:44 2019
GP error: ERROR 999999: Error executing function.
GP error: The table was not found. [VAT_flow_accumulation]
GP error: Failed to execute (ZonalStatisticsAsTable).
GP error: Failed at Mon Sep 30 15:00:44 2019 (Elapsed Time: 0.27 seconds)

The tool runs fine in 10.5.1 on this AOI. ZonalStatisticsAsTable runs fine in BAGIS under 10.7.1. We use it for generating the elevation precip table. The ZonalStatisticsAsTable GP tool seems to run fine with the flow_accumulation inputs when I run it manually under 10.7.1. The flow accumulation raster is continuous and doesn't have an attribute table.

Any ideas?

**Comment by lbross** [comment]
Dug into this some more this afternoon. Noticed that I was using a vector for the snap raster. Changed this to the aoi grid and it works fine! Not sure why the vector worked in AGS 10.5.1 or manually through the GP. Made this change to the source code for 10.5.1.
All of the libraries in BAGIS-H had the specific version property set to 'true'. After I fixed this glitch with the Assign Sequential HRU numbers tool, the 10.5.1 add-in seemed to work fine on 10.7.1. I published a new version of the add-in to basins in the 'For ArcGIS10.5.X and later' folder. The new version number is BAGIS_H_1.6.6_b3.esriAddIn.
If this version works for you, I may try republishing the BAGIS add-in from 10.5.1, but with the specific version property set to 'true'. It will be more efficient if we can compile the add in for 10.5.1 and have it work with 10.7.1.
Please let me know how testing BAGIS-H goes.

**Comment by jdduh** [comment]
Reposting Lesley's message from #26 :
"I posted a new BAGIS add-in to Basins FTP. After updating the 'Specific Version' property on all the ESRI libraries to 'No', I think we can use the same addin for both 10.5 and 10.7. It should work better on 10.6 than previous versions too. I tested it on my RDP computer that is running 10.7.1 and found no issues."

**Comment by lbross** [comment]
The upgrade for BAGIS-P has been more difficult than the others. The FileDialog controls that we use to select the templates crashed the program. After researching online, I set the AutoUpgradeEnabled property to false for these controls. Although the file browser isn't as attractive with this setting, it works. I also ran into the issue again with using a vector as the snap raster for the ZonalStatisticsAsTable in a couple of places but fixed that by changing to a grid snap raster.

However, the models do not run well under 10.7.1. I just spent about 30 minutes trying to get snow_intcp to run without success. Given the status of BAGIS-P, I'm not sure how much time I should be spending on this. I can post a new add-in for BAGIS-P that works on 10.7.1 if you want to take a look?
